python/ray/util/spark/cluster_init.py

```diff
@@ -136,7 +135,8 @@ def wait_until_ready(self):
                         "server cannot be found. They can be installed with "
                         "pip install ray[default]."
                     )
-            ray_session_dir = _global_node.get_session_dir_path()
+            # ray.init makes sure _global_node is not None
+            ray_session_dir = ray._private.worker._global_node.get_session_dir_path()
```

Comment:
`ray._private.worker._global_node.get_session_dir_path()` might not work if it is executed inside `start_ray_node` process.